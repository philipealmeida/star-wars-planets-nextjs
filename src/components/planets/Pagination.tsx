import styles from '@/styles/components/planets/Pagination.module.css';

interface PaginationProps {
  currentPage: number;
  totalPages: number;
  onPageChange: (page: number) => void;
}

export default function Pagination({
  currentPage,
  totalPages,
  onPageChange,
}: PaginationProps) {
  return (
    <div className={styles.pagination}>
      <button
        onClick={() => onPageChange(currentPage - 1)}
        disabled={currentPage === 1}
        className={styles.button}
      >
        Previous
      </button>
      <span className={styles.info}>
        Page {currentPage} of {totalPages}
      </span>
      <button
        onClick={() => onPageChange(currentPage + 1)}
        disabled={currentPage === totalPages}
        className={styles.button}
      >
        Next
      </button>
    </div>
  );
}