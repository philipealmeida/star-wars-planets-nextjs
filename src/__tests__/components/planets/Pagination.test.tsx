import Pagination from '@/components/planets/Pagination';
import { render, screen, fireEvent } from '@testing-library/react';

const renderComponent = (props: {
  currentPage: number;
  totalPages: number;
  onPageChange: (page: number) => void;
}) => {
  return render(<Pagination {...props} />);
};

describe('Pagination', () => {
  const mockOnPageChange = jest.fn();

  it('matches snapshot', () => {
    const { container } = renderComponent({
      currentPage: 1,
      totalPages: 5,
      onPageChange: mockOnPageChange,
    });
    expect(container).toMatchSnapshot();
  });

  it('disables previous button on first page', () => {
    renderComponent({
      currentPage: 1,
      totalPages: 5,
      onPageChange: mockOnPageChange,
    });
    
    const prevButton = screen.getByText('Previous');
    expect(prevButton).toBeDisabled();
  });

  it('disables next button on last page', () => {
    renderComponent({
      currentPage: 5,
      totalPages: 5,
      onPageChange: mockOnPageChange,
    });
    
    const nextButton = screen.getByText('Next');
    expect(nextButton).toBeDisabled();
  });

  it('calls onPageChange with correct values', () => {
    renderComponent({
      currentPage: 2,
      totalPages: 5,
      onPageChange: mockOnPageChange,
    });
    
    fireEvent.click(screen.getByText('Next'));
    expect(mockOnPageChange).toHaveBeenCalledWith(3);

    fireEvent.click(screen.getByText('Previous'));
    expect(mockOnPageChange).toHaveBeenCalledWith(1);
  });
});