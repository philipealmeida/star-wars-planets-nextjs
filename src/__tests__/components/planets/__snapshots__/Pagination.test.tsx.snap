// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination matches snapshot 1`] = `
<div>
  <div
    class="pagination"
  >
    <button
      class="button"
      disabled=""
    >
      Previous
    </button>
    <span
      class="info"
    >
      Page 
      1
       of 
      5
    </span>
    <button
      class="button"
    >
      Next
    </button>
  </div>
</div>
`;
